src/Engine/Graphics/IRender.h

```diff
@@ -149,6 +149,7 @@ struct RenderBillboardD3D {
     int PaletteIndex;
 };
 
+// TODO(pskelton): do we really need 3 different billboard structs
 struct SoftwareBillboard {
```

Comment:
There was software renderer code before, which got deleted.
Basically mm6 ran in full soft mode, then mm7 introduced DirectX mode . But looking at the code you can see that a lot of mm6 render related stuff was just duplicated with some/minimal change. It was all clearly rushed by the developers,  so it seems.
Anyways, yeah, be weary of anything with "Software" in it (as opposed to D3D) as this is how I makred the distinction to which renderer used which structure

Reply:
Yeah there's a few places now where I believe we can simplify/ remove the separate structs.